File: code/smv-debugger/src/main/java/org/fbme/smvDebugger/panel/mvc/GlobalTimeModel.java
package org.fbme.smvDebugger.panel.mvc;

import javax.swing.event.SwingPropertyChangeSupport;
import java.beans.PropertyChangeListener;

public class GlobalTimeModel {
  public static final String GLOBAL_TIME = "globalTime";

  private final SwingPropertyChangeSupport propertyChangeSupport = new SwingPropertyChangeSupport(this);
  private String time;

  public String getTime() {
    return time;
  }

  public void setTime(final String time) {
    final String oldTime = this.time;
    this.time = time;
    propertyChangeSupport.firePropertyChange(GLOBAL_TIME, oldTime, this.time);
  }

  public void addPropertyChangeListener(final PropertyChangeListener listener) {
    propertyChangeSupport.addPropertyChangeListener(listener);
  }

  public void addPropertyChangeListener(final String propertyName, final PropertyChangeListener listener) {
    propertyChangeSupport.addPropertyChangeListener(propertyName, listener);
  }
}

File: code/smv-debugger/src/main/kotlin/org/fbme/smvDebugger/panel/mvc/GlobalTimeModel.kt
package org.fbme.smvDebugger.panel.mvc

import java.beans.PropertyChangeListener
import javax.swing.event.SwingPropertyChangeSupport

class GlobalTimeModel {
    private val propertyChangeSupport = SwingPropertyChangeSupport(this)
    var time: String? = null
        set(time) {
            val oldTime = this.time
            field = time
            propertyChangeSupport.firePropertyChange(GLOBAL_TIME, oldTime, this.time)
        }

    fun addPropertyChangeListener(listener: PropertyChangeListener?) {
        propertyChangeSupport.addPropertyChangeListener(listener)
    }

    fun addPropertyChangeListener(propertyName: String?, listener: PropertyChangeListener?) {
        propertyChangeSupport.addPropertyChangeListener(propertyName, listener)
    }

    companion object {
        const val GLOBAL_TIME = "globalTime"
    }
}
